packages/cletus/src/operations/dba.tsx

```diff
@@ -604,6 +604,65 @@ export const data_index = operationOf<
   },
 });
 
+export const data_search = operationOf<
+  { name: string; query: string; n?: number },
+  { query: string; results: Array<{ source: string; text: string; similarity: number }> }
+>({
+  mode: 'read',
+  status: ({ name, query }) => `Searching ${name}: ${abbreviate(query, 25)}`,
+  analyze: async ({ name, query, n }, { config }) => {
+    const type = getType(config, name);
+    const limit = n || 10;
+    return {
+      analysis: `This will search ${type.friendlyName} knowledge for "${query}", returning up to ${limit} results.`,
+      doable: true,
+    };
+  },
+  do: async ({ name, query, n }, { ai }) => {
+    const knowledge = new KnowledgeFile();
+    await knowledge.load();
+
+    const limit = n || 10;
+    
+    // Generate embedding for query
+    const embeddingResult = await ai.embed.get({ texts: [query] });
+    const modelId = getModel(embeddingResult.model).id;
+    const queryVector = embeddingResult.embeddings[0].embedding;
+
+    // Search for similar entries with source prefix "type:"
+    const similarEntries = knowledge.searchBySimilarity(modelId, queryVector, limit);
+
```

Comment:
Search by similarity should have an optional sourcePrefix to narrow down how many computations it does.